src/features/envVisualizer/components/arrows/GenericArrow.tsx

```diff
@@ -48,25 +81,59 @@ export class GenericArrow implements Visible, Hoverable {
   protected calculateSteps(): StepsArray {
     const to = this.target;
     if (!to) return [];
-    return [(x, y) => [to.x, to.y]];
+    return [(x, y) => [to.x(), to.y()]];
   }
-
-  onMouseEnter = ({ currentTarget }: KonvaEventObject<MouseEvent>) => {
-    setHoveredStyle(currentTarget, {
+  getStrokeWidth(): number {
+    return Number(Config.ArrowStrokeWidth);
+  }
+  onMouseEnter(e: KonvaEventObject<MouseEvent>) {
+    setHoveredCursor(e.target);
+    setHoveredStyle(e.currentTarget, {
       strokeWidth: Number(Config.ArrowHoveredStrokeWidth)
     });
-  };
-
-  onMouseLeave = ({ currentTarget }: KonvaEventObject<MouseEvent>) => {
-    setUnhoveredStyle(currentTarget, {
-      strokeWidth: Number(Config.ArrowStrokeWidth)
-    });
-  };
-
+  }
+  onClick({ currentTarget }: KonvaEventObject<MouseEvent>) {
+    this.selected = !this.selected;
+    if (!this.isSelected()) {
+      if (
+        // !(this.source instanceof ArrayUnit && this.source.parent.isSelected()) &&
```

Comment:
unused code. (lines 118-119 too). please have a glance through the whole PR again. will not add further comments on this as it's getting long